projects/ngx-formentry/src/form-entry/helpers/js-expression-helper.ts

```diff
@@ -181,7 +181,9 @@ export class JsExpressionHelper {
     const bmiIdx = getBmiBin(bmi);
     const sbpIdx = 4 - getSbpBin(sbp);
 
-    return southEastAsiaCvdRiskTables[sexIdx][smokerIdx][ageIdx][sbpIdx][bmiIdx];
+    const hasAllAttributes = sex && smoker && age && sbp && bmi;
```

Comment:
A couple of things:

1. We should move this "missing values" check to the top of the function, so we don't do any of the other conversions before this.
2. We need to be careful with using `&&` and `!` here since this will actually misbehave with value data (what happens if someone is *not* a smoker?).

So something like:

```ts
if ((sex === null || typeof sex === "undefined") && (smoker === null || typeof smoker === "undefined") // etc.) {
  return null;
}

// function as usual
```

Reply:
Thanks @ibacher I will change that.

Reply:
@ibacher One thing I was wondering, is why isn't typescript flagging when some values are not passed. because we are attaching type checks to the values.

Reply:
@hadijahkyampeire Typescript only exists at the source-code level. In production, there are no type checks from type script. So the type checks will prevent you from explicitly doing something like:

```ts
// void 0 is what undefined gets compiled to because void 0 === undefined
calcSouthEastAsiaNonLabCVDRisk(void 0, void 0, void 0, void 0, void 0);
```

However, at runtime, it's just Javascript so where you need to validate runtime types (like we do here), we need a runtime check. If you want to see the code that gets emitted you can use the Typescript playground ([here is an example of the CVD risk score](https://www.typescriptlang.org/play?#code/ATBmFcDsGMBcEsD2ljQIYBtoGVHlgBYCiaAzrAIKnxoByyAMmgEYDCAagCIBK8pA1gApSAUwAeALmAByALLTgAHxkAxaQBpgpALaJ+IgE5TmiRBhFpImtAHMRUyOG3NDm0swAODpy4Obm2vDezoYAlMAA3gBQICAA9HHAAELwKBAwCMikMbHQWbDAdpR2KSgAvMCCtiLhZQB8wLJohAB0oBimBoKCTa2BkD3NBC3aaGKCACwADNZ2oZoA7BPhALTA0+GJAKyhANw5IHmQ5IUisNiepcAVwp61Db3Do+MzjUNtHYhdg32ptx6aACMAA4pqtgICAExg4CJaHhADUEL2B1Q+VOsCSgSuNwC8Hubz6Y0Er0eH063TJ-UEeM0AGYdsA1vDYcBGUjASjUQlgOw0AYaMxzNlYmjjgVRGIAJIAEzE1y04muZQqcgUAH5gFNgFJAftRUcTjo9IZZfKKsb9AZgJrATqtfrcujqmaFQA2JkYiglP7VFEG9F410VIpY+ClGmBf1O8VaTzB9aeooXDwR9weLkB2MEMgUDAYCiwWAC5j4ESkBWS4AAMmrWl0VprdeqTbjHlbeP2qIMZ3ABhQAEIc6Q8wWiyWyxXNY58-bSHhCCRyFQaKwAG4y3gCAAqLGFAG1JWaALqHhumuWnl2Xw-xm9By+OgC+USAA)).

Reply:
The reason for using Typescript is to try to prevent programmer errors, but it doesn't help with all type errors because, at the end of the day, it's just Javascript.

Reply:
Gotcha, thanks @ibacher.

Also, I have used the `||` instead of `&&` because we want it to check if any of the 5 values are not passed, and `&&` was passing if one of them is passed. it would be `false && true && true...` which results to `false` But with `||` if any value check returns `true` because it is undefined or null, then our check will be `true` thus returning `null`.